src/components/DataInput/CheckboxInput.tsx

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import React from "react";
+import { Checkbox, FormControl, FormControlLabel, FormGroup, FormLabel } from "@mui/material";
+
+interface Props {
+    optionLabels: string[];
+    optionKeys: string[];
+    groupLabel?: string;
```

Comment:
For a label you could use props.children to add flexibility and clarity in downstream markup